File: LeetCode/FirstUniqueCharacter.java
class Solution {
    public int firstUniqChar(String s) {  
        // create a hash map to count instances using chacters of our string as keys
        HashMap<Character, Integer> map = new HashMap<>();
        for(int i = 0; i < s.length(); i++){
          // use the charAt() method to get our key
          char ch = s.charAt(i);
          // add our key and counter to the map
          map.put(ch, map.getOrDefault(ch, 0) + 1);
        }
        // reiterate through our map
        for(int i = 0; i < s.length(); i++)
          // if our the value of any of our key is 1, return that index
          if(map.get(s.charAt(i)) == 1)
            return i;
        // it was not found so return -1
        return -1;
    }
}

File: LeetCode/firstUniqueCharacter.js
var firstUniqChar = function(s) {
    // create a map object
    let map = {};
    // iterate through our string adding each character of our string as a property
    for(let i = 0; i < s.length; i++){
        let ch = s.charAt(i);
        // if the property exists add 1, else create it and itialize it to 1
        map[ch] = (map[ch] || 0) + 1;
    }
    
    // iterate through our string using the characters as our property calls
    for(let i = 0; i < s.length; i++)
        // if the value of the property is 1 we return the index of the string
        if(map[s.charAt(i)] === 1)
            return i;
    // return -1 if not found
    return -1;
};
